### PR TITLE
Error dialogs

### PR DESCRIPTION
**Pull request by d-loose**
WIP
- add generalized dialog box
- add error dialog
- add error state
- show error dialog on `FwupdException` during `install()`
- add widget tests for dialogs
- add localized error messages
image

### CODE REVIEW

File: lib/src/widgets/confirmation_dialog.dart [hunk 1]
```diff
@@ -28,23 +30,20 @@ Future<void> showConfirmationDialog(
         constraints: const BoxConstraints(maxWidth: 500),
         child: Row(
           children: [
-            const Icon(
-              YaruIcons.question,
-              size: 64.0,
-            ),
+            if (icon != null) icon,
             const SizedBox(width: 16),
```
Comment (jpnurmi) [inline lib/src/widgets/confirmation_dialog.dart:34]: do we want the spacing if there's no icon or should it be inside the icon != null check?

**Review by jpnurmi — COMMENTED** [review]
LGTM otherwise but I wonder if we could make the button outlined?

File: lib/src/widgets/confirmation_dialog.dart (hunk 1 above)
Comment (d-loose) [reply lib/src/widgets/confirmation_dialog.dart]: Good catch, didn't see that - the space should only be there if there's also an icon.

**Review by jpnurmi — APPROVED** [review]
LGTM :+1:

File: lib/src/l10n/app_en.arb [hunk 2]
```diff
@@ -54,7 +54,27 @@
...
+  "fwupdErrorAlreadyPending": "Already set to be installed offline",
+  "fwupdErrorAuthFailed": "Failed to get authentication",
+  "fwupdErrorRead": "Failed to read from device",
+  "fwupdErrorWrite": "Failed to write to the device",
+  "fwupdErrorInvalidFile": "Invalid file format",
+  "fwupdErrorNotFound": "No matching device exists",
+  "fwupdErrorNothingToDo": "Nothing to do",
+  "fwupdErrorNotSupported": "Action was not possible",
+  "fwupdErrorSignatureInvalid": "Signature was invalid",
+  "fwupdErrorAcPowerRequired": "ac power was required",
```
[the remark this answers is not in the document]
Comment (d-loose) [reply lib/src/l10n/app_en.arb:69]: Yes, that must've been a vim macro mishap from when I parsed the list of error messages from #53 :laughing:

**Pull request merged**